### ISSUE TITLE
Define interface.IProcessProtocol

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3020]
|<img alt="bigdog's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| bigdog reported|
|-|-|
|Trac ID|trac#3020|
|Type|enhancement|
|Created|2008-02-04 18:24:12Z|
|Branch|https://github.com/twisted/twisted/tree/process-interface-3020|

define   interface.IProcessProtocol that defines the API for Twisted Processes.

Attachments:

* [protocol.py_diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/387/38740b4e138067226f9545bf27346eda810207ea/571eecc6c98e8b62e89e24d8356ebd10149a44ed.py_diff) (416 bytes) - added by bigdog on 2008-02-04 21:55:56Z - add implments(IProcessProtocol) to ProcessProtocol
* [interfaces.py_patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/387/38740b4e138067226f9545bf27346eda810207ea/3ae427b51c429b8e01bbfda37e0ffc30d4b9fc95.py_patch) (2271 bytes) - added by bigdog on 2008-02-04 22:32:43Z - 
* [test_process.py.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/387/38740b4e138067226f9545bf27346eda810207ea/531108510b454e15ea3bd6fee73e2efa139c0f86.diff) (572 bytes) - added by bigdog on 2008-02-05 01:26:53Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3020 3020
type__enhancement enhancement
reporter__bigdog bigdog
priority__normal normal
milestone__ 
branch__branches_process_interface_3020 branches/process-interface-3020
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1202149452000000 1202149452000000
changetime__1203621521000000 1203621521000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="bigdog's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|bigdog set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="bigdog's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|bigdog commented|
|-|-|

Add Comments to interfaces.py to address ticket 2852

**Comment by twisted-trac** [comment]
|<img alt="bigdog's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|bigdog removed owner<br>bigdog set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [22491]) Branching to 'process-interface-3020'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

Applied the patches, made some changes.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_7"></a>|@dreid set owner to @exarkun|
|-|-|

This is great bigdog/exarkun please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [22679]) Merge process-interface-3020

Author: bigdog, exarkun
Reviewer: dreid
Fixes [#3020](https://github.com/twisted/twisted/issues/3020)
Refs [#2852](https://github.com/twisted/twisted/issues/2852)

Define an interface for process protocols, twisted.internet.interfaces.IProcessProtocol.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
